src/Logging.W3C/src/W3CLogger.cs

```diff
@@ -0,0 +1,166 @@
+using System;
+using Microsoft.Extensions.Options;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using System.Numerics;
+using System.Text;
+using Microsoft.Net.Http.Headers;
+using System.Text.RegularExpressions;
+using System.Globalization;
+
+namespace Microsoft.Extensions.Logging.W3C
+{
+    internal sealed class W3CLogger : ILogger, IDisposable
+    {
+        private readonly string _name;
+        private readonly W3CLoggerProcessor? _messageQueue;
+        private readonly IOptionsMonitor<W3CLoggerOptions> _options;
+        private readonly bool _isActive;
+        private readonly W3CLoggingFields _loggingFields;
+
+        internal W3CLogger(string name, IOptionsMonitor<W3CLoggerOptions> options)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            _name = name;
+            _options = options;
+
+            // If the info isn't coming from the _w3cLogger in HttpLoggingMiddleware, don't log anything
+            if (name == "Microsoft.AspNetCore.W3CLogging")
+            {
+                _isActive = true;
+                _messageQueue = new W3CLoggerProcessor(_options);
+                _loggingFields = _options.CurrentValue.LoggingFields;
+                LogFileFullName = _messageQueue.FullName;
+            }
+        }
+
+        public string? LogFileFullName { get; }
+
+        // TODO - do we need to do anything here?
+        public IDisposable BeginScope<TState>(TState state)
```

Comment:
TODO - do I need to do anything here? Logging.AzureAppServices does the same (without disabling CS8603): https://github.com/dotnet/aspnetcore/blob/be26b8fc70c7a43419307b6aec6b61a1aff0e3bd/src/Logging.AzureAppServices/src/BatchingLogger.cs#L22-L24